src/Magento/FunctionalTestingFramework/Test/etc/Actions/customActions.xsd

```diff
@@ -29,6 +30,25 @@
 
     <!-- Complex Types -->
 
+    <xs:complexType name="helperType">
+        <xs:sequence>
+            <xs:element name="argument" type="argumentType" minOccurs="0" maxOccurs="unbounded" />
+        </xs:sequence>
+        <xs:attribute type="xs:string" name="class" use="required" />
+        <xs:attribute type="xs:string" name="method" use="required" />
+        <xs:attributeGroup ref="commonActionAttributes"/>
+    </xs:complexType>
+
+    <xs:complexType name="argumentType" mixed="true">
+        <xs:attribute name="name" use="required">
+            <xs:simpleType>
+                <xs:restriction base="xs:string">
+                    <xs:pattern value="[^cm].*|c(.{0,3}|[^l].*|l[^a].*|la[^s].*|las[^s].*|lass.+)|m(.{0,4}|[^e].*|e[^t].*|et[^h].*|eth[^o].*|etho[^d].*|ethod.+)" />
```

Comment:
What does this regex do?

Reply:
verifies that you can not specify argument name "class" or "method" because they are reserved.

Reply:
But why is it so complex? I'm missing something.

Reply:
XSD version="1.0"  not allowing usage of negative lookahead 
that's the only way I was able to make it. Looks complex (or I call it ugly), but I wasted couple hours to make it work 😞

Reply:
gotcha. crazy!